File: components/commands/OutNode.py
from components.commands.Node import Node
from typing import Tuple


class OutNode(Node):

    def execute(self):
        """
        Returns received output.
        """
        return self.params

    def add_params(self, operand):
        """
        Add a parameter to the output node.
        """
        self.params.append(operand)

    def __init__(self, loc: Tuple[float, float]):
        self.params = []
        super().__init__(loc, './assets/triangle.png')

File: src/graph/OutNode.js
import Node from 'Node.js';

export default class OutNode extends Node{
    /**
     * Last node that stores the graph's output.
     * @param {int} numPorts Number of input ports the OutNode can take.
     */
    constructor(numPorts){
        super(numPorts);
    }
    addParams(operand){
        /**
         * Adds operand to output.
         * @param {Operand} operand Operand to add to output.
         */
        if (this.params.length < this.numPorts) {
            this.params.push(operand);
        }
    }
    execute(){
        /**
         * Returns output.
         */
        return this.params;
    }
}
